kotlin-mbedtls-netty/src/main/kotlin/org/opencoap/ssl/netty/DtlsChannelHandler.kt

```diff
@@ -90,6 +90,14 @@ class DtlsChannelHandler @JvmOverloads constructor(
 
     override fun write(ctx: ChannelHandlerContext, msg: Any, promise: ChannelPromise) {
         when (msg) {
+            is DatagramPacketWithContext -> {
+                write(msg, promise, ctx)
+                promise.toCompletableFuture().whenComplete { isSuccess, err ->
+                    if (msg.sessionContext.sessionExpirationHint && err == null && isSuccess) {
+                        dtlsServer.closeSession(msg.recipient())
+                    }
+                }
```

Comment:
Minor thing, but maybe we should check first if `msg.sessionContext.sessionExpirationHint` is true, and then add handler.
Instead of `.whenComplete` use `.thenAccept`